drivers/hmis/spec/requests/hmis/enrollment_services_visibility_spec.rb

```diff
@@ -0,0 +1,68 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  before(:all) do
+    cleanup_test_environment
+  end
+  after(:all) do
+    cleanup_test_environment
+  end
+
+  include_context 'hmis base setup'
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+  let!(:c1) { create :hmis_hud_client, data_source: ds1 }
+  let!(:e1) { create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c1 }
+  let!(:s1) { create :hmis_hud_service, data_source: ds1, enrollment: e1, client: c1 }
+
+  # override base setup
+  let!(:cst1) { create :hmis_custom_service_type_for_hud_service, data_source: ds1, custom_service_category: csc1, user: u1 }
+
+  # canary values
+  let!(:p2) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1 }
+  let!(:e2) { create :hmis_hud_enrollment, data_source: ds1, project: p2, client: c1 }
+  let!(:s2) { create :hmis_hud_service, data_source: ds1, enrollment: e2, client: c1 }
+
+  before(:each) do
+    hmis_login(user)
+  end
+
+  describe 'enrollment services query' do
+    let(:query) do
+      <<~GRAPHQL
+        query TestQuery($id: ID!) {
+          enrollment(id: $id) {
+            id
+            services(limit: 10, offset: 0) {
+              nodes {
+                id
+              }
+            }
+          }
+        }
+      GRAPHQL
+    end
+
+    it 'resolves only services for this enrollment' do
```

Comment:
This looks good to me. I can think of two things to give it more coverage, which may be worth doing:
1. Give `hmis_user` access to view the p2 project, but not the p2 enrollment details. They still should not be able to 
view the p2 services. Something like:
```
# user can see enrollment details in p1
create_access_control(hmis_user, p1, with_permission: [:can_view_clients, :can_view_project, :can_view_enrollment_details])
# user can not see enrollment details in p2
create_access_control(hmis_user, p2, with_permission: [:can_view_clients, :can_view_project])
``

2. Add a CustomService, so we are ensuring correct access for both HUD and Custom service

Reply:
Good idea, thanks!

I built out enrollment_access_spec with tests for services and households that mirror those for enrollments. I figure that is okay since they are enrollment related but could also move to dedicated spec.